static/app/components/replays/breadcrumbs/breadcrumbItem.tsx

```diff
@@ -45,41 +48,95 @@ import {makeFeedbackPathname} from 'sentry/views/userFeedback/pathnames';
 type MouseCallback = (frame: ReplayFrame, nodeId?: number) => void;
 
 interface Props {
+  allowShowSnippet: boolean;
   frame: ReplayFrame;
   onClick: null | MouseCallback;
   onInspectorExpanded: OnExpandCallback;
   onMouseEnter: MouseCallback;
   onMouseLeave: MouseCallback;
+  onShowSnippet: () => void;
+  showSnippet: boolean;
   startTimestampMs: number;
   className?: string;
   expandPaths?: string[];
   extraction?: Extraction;
   ref?: React.Ref<HTMLDivElement>;
   style?: CSSProperties;
+  updateDimensions?: () => void;
 }
 
 function BreadcrumbItem({
   className,
-  extraction,
   frame,
   expandPaths,
   onClick,
   onInspectorExpanded,
   onMouseEnter,
   onMouseLeave,
+  showSnippet,
   startTimestampMs,
   style,
   ref,
+  onShowSnippet,
+  updateDimensions,
+  allowShowSnippet,
 }: Props) {
   const {color, description, title, icon} = getFrameDetails(frame);
   const {replay} = useReplayContext();
+  const organization = useOrganization();
+  const {data: extraction, isPending} = useExtractDomNodes({
+    replay,
+    frame,
+    enabled: showSnippet,
+  });
+
+  const prevExtractState = useRef(isPending);
+
+  useEffect(() => {
+    if (!updateDimensions) {
+      return;
+    }
+
+    if (isPending !== prevExtractState.current || showSnippet) {
+      prevExtractState.current = isPending;
+      updateDimensions();
+    }
+  }, [isPending, updateDimensions, showSnippet]);
+
+  const handleViewHtml = useCallback(
+    (e: React.MouseEvent<HTMLAnchorElement>) => {
+      onShowSnippet();
+      e.preventDefault();
+      e.stopPropagation();
+      trackAnalytics('replay.view_html', {
+        organization,
+        breadcrumb_type: 'category' in frame ? frame.category : 'unknown',
+      });
+    },
+    [onShowSnippet, organization, frame]
+  );
 
   const renderDescription = useCallback(() => {
     return typeof description === 'string' ||
       (description !== undefined && isValidElement(description)) ? (
-      <Description title={description} showOnlyOnOverflow isHoverable>
-        {description}
-      </Description>
+      <DescriptionWrapper>
+        <Description title={description} showOnlyOnOverflow isHoverable>
+          {description}
+        </Description>
+
+        {allowShowSnippet &&
+          !showSnippet &&
+          frame.data?.nodeId !== undefined &&
+          (!isSpanFrame(frame) || !isWebVitalFrame(frame)) && (
+            <ViewHtmlLink
+              to=""
+              disabled={showSnippet && isPending}
```

Comment:
won't need `showSnippet` in the `disabled` prop because `!showSnippet` is part of the condition to render on line 128